schedview/app/prenight/prenight.py

```diff
@@ -885,8 +942,17 @@ def prenight_app(*args, **kwargs):
     except KeyError:
         data_from_urls = False
 
+    try:
+        data_from_archive = kwargs["data_from_archive"]
+        del kwargs["data_from_archive"]
+    except KeyError:
+        data_from_archive = False
+
     if data_from_urls:
         prenight = Prenight()
+    elif data_from_archive:
+        resource_uri = kwargs["resource_uri"] if "resource_uri" in kwargs else None
+        prenight = ArchiveInputPrenight(resource_uri=resource_uri)
```

Comment:
You can, in fact, load data from an arbitrary external URL if you use the data_from_urls option. This is what makes this approach a security issue if it's run on shared infrastructure.

Reply:
An archive is not any old URI, but the address of the resource with the archive. It needs to have the expected directory layout (subdirecties with a given format and metadata yaml files with a given format in them), but it doesn't need to be an S3 bucket. I've been testing it on local disk with local files with URI's that look like `file:///my/test/archive/dir`. In principle, it could located at any URI that `lsst.resources` supports.

Reply:
I suspect it might not work with an http or https URI, though, because lsst.resources does not implement the whole API for http, and the code uses a part of the API that throws a NotImplementedError. So, only some kinds of URIs are supported.

Reply:
Ok, thanks for explaining. I suspect some additional docstrings or comments in the code would help still.